### PR TITLE
buzztrax: init at unstable-2022-01-26

### PR DESCRIPTION
**Pull request by bendlas**
###### Description of changes

Added package for https://www.buzztrax.org/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
> If your PR has been sitting for a while and you’d like to see some action on it please post it here!

I guess 5 minutes doesn't really fit that definition. The package even still has evaluation failures.

File: pkgs/applications/audio/buzztrax/default.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+, automake, autoconf, gtk-doc, intltool, libtool, pkgconfig, yelp-tools, itstool
+, gst_all_1
+, glib, libgsf, libxml2
+, clutter-gtk, gtk2
+, orc, fluidsynth
+}:
+
+stdenv.mkDerivation {
+
+  name = "buzztrax-git";
```
Comment (wegank) [inline pkgs/applications/audio/buzztrax/default.nix:11]:
```suggestion
  pname = "buzztrax";
  version = "unstable-2022-01-26";
```

File: pkgs/applications/audio/buzztrax/default.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Buzztrax";
+    repo = "buzztrax";
+    rev = "833287c6a06bddc922cd346d6f0fcec7a882aee5";
+    sha256 = "sha256-iI6m+zBWDDBjmeuU9Nm4aIbEKfaPe36APPktdjznQpU=";
```
Comment (wegank) [inline pkgs/applications/audio/buzztrax/default.nix:17]:
```suggestion
    hash = "sha256-iI6m+zBWDDBjmeuU9Nm4aIbEKfaPe36APPktdjznQpU=";
```

File: pkgs/applications/audio/buzztrax/default.nix [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  '';
+
+  NIX_CFLAGS_COMPILE = [
+    ## 'g_memdup' is deprecated: Use 'g_memdup2' instead
+    "-Wno-error=deprecated-declarations"
+  ];
+
+  # FONTCONFIG_FILE = makeFontsConf {
+  #   fontDirectories = [ ];
+  # };
```
Comment (wegank) [inline pkgs/applications/audio/buzztrax/default.nix:31]:
```suggestion
```

File: pkgs/applications/audio/buzztrax/default.nix [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-plugins-ugly
+    orc
+    fluidsynth
+  ];
+
+  meta = with lib; {
+    description = "Buzztrax is a modular music composer for Linux.";
+    homepage = "https://www.buzztrax.org/";
+    license = licenses.lgpl2;
```
Comment (wegank) [inline pkgs/applications/audio/buzztrax/default.nix:61]:
```suggestion
    license = licenses.lgpl21Plus;
```

File: pkgs/applications/audio/buzztrax/default.nix [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+  configureScript = "./autogen.sh";
+  configureFlags = [
+    "--enable-debug=no"
+  ];
+
+  nativeBuildInputs = [
+    automake autoconf gtk-doc intltool libtool pkgconfig
+    wrapGAppsHook yelp-tools itstool
+  ];
```
[the remark this answers is not in the document]
Comment (bendlas) [reply pkgs/applications/audio/buzztrax/default.nix:37]: Is disabling debug supposed to be part of `autoreconfHook`?

File: pkgs/applications/audio/buzztrax/default.nix (hunk 4 above)
Comment (wegank) [reply pkgs/applications/audio/buzztrax/default.nix:52]: `lgpl21` is deprecated in nixpkgs. The default substitute is `lgpl21Plus`, unless `lgpl21Only` is made explicit by upstream.

**Comment by wegank** [comment]
@ofborg build buzztrax buzztrax.passthru.tests

**Pull request merged**